packages/clay-date-picker/src/index.tsx

```diff
@@ -344,36 +344,59 @@ const ClayDatePicker: React.FunctionComponent<IProps> = React.forwardRef<
 		const inputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
 			const {value} = event.target;
 
-			const format = time
-				? `${dateFormat} ${use12Hours ? TIME_FORMAT_12H : TIME_FORMAT}`
-				: dateFormat;
+			if (!value) {
+				const newDaysSelected: [Date, Date] = [NEW_DATE, NEW_DATE];
 
-			const [startDate, endDate] = fromStringToRange(
-				value,
-				format,
-				NEW_DATE
-			);
+				const initialTime = {
+					hours: 0,
+					minutes: 0,
+				};
```

Comment:
Just a suggestion, just for cases like this where you won't change the variable later, you can pass these values directly to `setCurrentTime` this avoids the allocation of a new variable. You can probably do the same for `newDaysSelected`.